jax_galsim/image.py

```diff
@@ -1152,10 +1162,10 @@ def Image_isub(self, other):
         a = other
         dt = type(a)
     if dt == self.array.dtype:
-        array = self.array - a
+        self._array = self.array - a
     else:
-        array = (self.array - a).astype(self.array.dtype)
-    return Image(array, bounds=self.bounds, wcs=self.wcs)
+        self._array = (self.array - a).astype(self.array.dtype)
+    return self
```

Comment:
I think you only need to return `self` in methods that start with `i` ? like `iadd`

Reply:
e.g. Galsim `__sub__` returns a new image: https://github.com/GalSim-developers/GalSim/blob/a1f0a11a24690a279abd7027b116ec3e4986643d/galsim/image.py#L1644

Reply:
This one is for the isub method. The github diff is showing the wrong class.

Reply:
ah gotcha, thanks!